pkg/operator/ceph/csi/spec.go

```diff
@@ -311,5 +316,49 @@ func StartCSIDrivers(namespace string, clientset kubernetes.Interface, ver *vers
 			return fmt.Errorf("failed to create rbd service: %+v\n%+v", err, cephfsService)
 		}
 	}
+
+	if ver.Major > KubeMinMajor || (ver.Major == KubeMinMajor && ver.Minor >= provDeploymentSuppVersion) {
+		err = createCSIDriverInfo(clientset, RBDDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", RBDDriverName, err)
+		}
+		err = createCSIDriverInfo(clientset, CephFSDriverName)
+		if err != nil {
+			return fmt.Errorf("failed to create CSI driver object for %q: %+v", CephFSDriverName, err)
+		}
+	}
 	return nil
 }
+
+// createCSIDriverInfo Registers CSI driver by creating a CSIDriver object
+func createCSIDriverInfo(clientset kubernetes.Interface, name string) error {
+	var err error
+	attach := false
+	mountInfo := false
+	// Create CSIDriver object
+	csiDriver := &k8scsi.CSIDriver{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Spec: k8scsi.CSIDriverSpec{
+			AttachRequired: &attach,
+			PodInfoOnMount: &mountInfo,
+		},
+	}
+	csidrivers := clientset.StorageV1beta1().CSIDrivers()
+
+	err = wait.PollImmediate(2*time.Second, 15*time.Second, func() (done bool, err error) {
```

Comment:
Why do we need to retry this? Most commonly, creating a resource should succeed on the first try.

Reply:
removed, it's not needed